---
title: "LDL"
author: "Angela Andaleon"
output: html_document
---

##In RStudio

```{r LDL}
data <- read.table("/home/angela/px_yri_chol/GWAS/Phenotypes/YRI_phenotypes.txt", header = T)
hist(data$LDL)
	##Saved to "/home/angela/px_yri_chol/GWAS/LDL/hist_LDL.png"
shapiro.test(data$LDL)
	##"W = 0.98808, p-value = 2.18E-07"
	##For log10, "0.97228, p-value = 4.54e-13"
```

##In command line

```{bash pheno}
plink --bfile /home/angela/px_yri_chol/QC/QCStep6/QCStep6k/QCStep6k --maf 0.05 --linear --adjust --pheno /home/angela/px_yri_chol/GWAS/Phenotypes/YRI_phenotypes.txt --pheno-name LDL --out /home/angela/px_yri_chol/GWAS/LDL/LDL
  ##1522836 variants loaded from .bim file.
  ##1157 people (397 males, 760 females) loaded from .fam.
  ##961 phenotype values present after --pheno.
  ##Before main variant filters, 0 founders and 1157 nonfounders present.
  ##Total genotyping rate is 0.999242.
  ##0 variants removed due to minor allele threshold(s)
  ##(--maf/--max-maf/--mac/--max-mac).
  ##1522836 variants and 1157 people pass filters and QC.
  ##Writing linear model association results to /home/angela/px_yri_chol/GWAS/LDL/LDL.assoc.linear ... done.
  ##--adjust: Genomic inflation est. lambda (based on median chisq) = 1.
  ##--adjust values (1504671 variants) written to /home/angela/px_yri_chol/GWAS/LDL/LDL.assoc.linear.adjusted .
```

##In RStudio

```{r topSNPs}
LDL_linear <- read.table("/home/angela/px_yri_chol/GWAS/LDL/LDL.assoc.linear", header = T)
LDL_linear_noNA <- LDL_linear[complete.cases(LDL_linear),]
```
![](/home/angela/px_yri_chol/GWAS/LDL/LDL_topSNPs.png)
```{r qq}
source('/home/angela/px_yri_chol/GWAS/qqman.r')
qq(LDL_linear_noNA$P)
	##Saved to "/home/angela/px_yri_chol/GWAS/LDL/LDL_qq.png"
```
![](/home/angela/px_yri_chol/GWAS/LDL/LDL_qq.png)
```{r man}
dev.off()
source('/home/angela/px_yri_chol/GWAS/qqman.r')
manhattan(LDL_linear_noNA)
	##Saved to "/home/angela/px_yri_chol/GWAS/LDL/LDL_man.png"
```
![](/home/angela/px_yri_chol/GWAS/LDL/LDL_man.png)
```{r dev.off}
dev.off()
```